### PR TITLE
Add cache option

### PR DESCRIPTION
**Pull request by wfeller**
Is it useful to support multiple cache drivers like Laravel or would it be enough to simply only use the memory cache?

Solves #51

### CODE REVIEW

**Comment by iksaku** [comment]
I think using a Generic Cache adapter could suffice... Lets hear what @sausin thinks about this

**Comment by sausin** [comment]
Loving it!! I think the generic adapter looks good. We're primarily a laravel package but no harm in making things more generic if it's possible :+1: 

@wfeller Could you consider some tests for it?

**Comment by sausin** [comment]
Also just noticed, shouldn't the new dependencies be in `require` instead of `require-dev` ?

File: src/Commands/SetTempUrlKey.php [hunk 1]
```diff
@@ -48,7 +49,13 @@ class SetTempUrlKey extends Command
     public function handle(): void
     {
         try {
-            $this->container = Storage::disk($this->option('disk'))->getAdapter()->getContainer();
+            $adapter = Storage::disk($this->option('disk'))->getAdapter();
+
+            if ($adapter instanceof CachedAdapter) {
+                $adapter = $adapter->getAdapter();
```
Comment (sausin) [inline src/Commands/SetTempUrlKey.php:55]: There are two calls to `getAdapter`. Is this correct?

File: composer.json [hunk 2]
```diff
@@ -34,7 +34,9 @@
         "phpunit/phpunit": "^7.0",
         "mockery/mockery": "^1.0",
         "friendsofphp/php-cs-fixer": "^2.16.1",
-        "brainmaestro/composer-git-hooks": "^2.8"
+        "brainmaestro/composer-git-hooks": "^2.8",
+        "league/flysystem-cached-adapter": "^1.0",
```
Comment (sausin) [inline composer.json:38]: These should probably be under `require`

**Comment by wfeller** [comment]
No, the other dependencies are only required if the users want to use the cache features, in which case they can install them themselves (I'll add a suggest part in composer.json like Laravel does it - see https://github.com/laravel/framework/blob/1dc032a15037330c5065b84ce3937d075a673b5d/composer.json#L130).

I'll update the PR to only support memory cache + tests tonight or tomorrow.

**Comment by sausin** [comment]
I see, makes sense. We should reflect the same in the config documentation as well :+1:

File: src/Commands/SetTempUrlKey.php (hunk 1 above)
Comment (wfeller) [reply src/Commands/SetTempUrlKey.php:55]: Yes, same as https://github.com/illuminate/filesystem/blob/8eaa230711347ac76b3b61372d1b3b332e0931ee/FilesystemAdapter.php#L426

I'll make sure this works as intended when I write tests.

Comment (sausin) [reply src/Commands/SetTempUrlKey.php:55]: Excellent! Thanks :+1:

**Comment by wfeller** [comment]
@sausin @iksaku Please take a look at the PR and let me know if you think anything should be updated, otherwise I believe it's good to go.

**Comment by sausin** [comment]
Looks good to me. Your thoughts @iksaku ?

**Pull request merged**